fakeable_test.go

```diff
@@ -0,0 +1,81 @@
+package gofakeit
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+type strTyp string
+
+func (t strTyp) Fake(faker *Faker) interface{} {
+	return faker.FirstName()
+}
+
+type strTypPtr string
+
+func (t *strTypPtr) Fake(faker *Faker) interface{} {
+	return strTypPtr("hello test ptr")
+}
+
+type testStruct1 struct {
+	B string `fake:"{firstname}"`
+}
+
+type testStruct2 struct {
+	B strTyp
+}
+
+func TestIsFakeable(t *testing.T) {
+	var t1 testStruct2
+	var t2 *testStruct2
+	var t3 strTyp
+	var t4 *strTyp
+	var t5 strTypPtr
+	var t6 *strTypPtr
+
+	if isFakeable(reflect.ValueOf(t1).Type()) {
+		t.Errorf("expected testStruct2 not to be fakeable")
+	}
+
+	if isFakeable(reflect.ValueOf(t2).Type()) {
+		t.Errorf("expected *testStruct2 not to be fakeable")
+	}
+
+	if !isFakeable(reflect.ValueOf(t3).Type()) {
+		t.Errorf("expected strTyp to be fakeable")
+	}
+
+	if !isFakeable(reflect.ValueOf(t4).Type()) {
+		t.Errorf("expected *strTyp to be fakeable")
+	}
+
+	if !isFakeable(reflect.ValueOf(t5).Type()) {
+		t.Errorf("expected strTypPtr to be fakeable")
+	}
+
+	if !isFakeable(reflect.ValueOf(t6).Type()) {
+		t.Errorf("expected *strTypPtr to be fakeable")
+	}
+}
+
+func ExampleFakeable() {
```

Comment:
Just to help give the best examples of this feature as possible can you throw in a couple of other custom examples to showcase what others can do with simple custom variables as well as structs?